tests/Export/Adapters/AttributeAdapterTest.php

```diff
@@ -0,0 +1,830 @@
+                'currencyId' => Defaults::CURRENCY,
+                'snippetSet' => [
+                    'name' => 'oof',
+                    'baseFile' => 'de.json',
+                    'iso' => 'de_AT'
+                ]
+            ]
+        ], Context::createDefaultContext());
+
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('url', $fullDomain));
+
+        // Wait until the domain entity has been created, since DAL works with events and these events
+        // run asynchronously on a different thread.
+        do {
+            $result = $domainRepo->search($criteria, Context::createDefaultContext());
+        } while ($result->getTotal() <= 0);
+
+        // The sales channel should use the newly generated URL instead of the default domain.
+        $this->salesChannelContext->getSalesChannel()->setLanguageId($result->getEntities()->first()->getLanguageId());
+        $this->salesChannelContext->getSalesChannel()->setDomains($result->getEntities());
+
+        $productEntity = $this->createTestProduct();
+
+        $config = $this->getMockedConfig();
+        $adapter = $this->getAttributeAdapter($config);
+        $attributes = $adapter->adapt($productEntity);
+
+        $hasSeoCatUrls = false;
+        foreach ($attributes as $attribute) {
+            if ($attribute->getKey() === 'cat_url') {
+                foreach ($attribute->getValues() as $value) {
+                    // We only care about SEO URLs of categories. Non-SEO categories are automatically generated
+                    // by the Shopware router.
+                    if (!(strpos($value, $catUrlWithoutSeoUrlPrefix) === 0)) {
+                        $hasSeoCatUrls = true;
+                        $this->assertStringStartsWith($expectedPath, $value);
+                    }
+                }
+            }
+        }
+
+        $this->assertTrue($hasSeoCatUrls);
+    }
+
+    private function getMockedConfig(string $integrationType = 'Direct Integration'): Config
+    {
+        $override = [
+            'languageId' => $this->salesChannelContext->getSalesChannel()->getLanguageId(),
+            'salesChannelId' => $this->salesChannelContext->getSalesChannel()->getId()
+        ];
+
+        return $this->getFindologicConfig($override, $integrationType === 'Direct Integration');
+    }
+
+    private function getAttributeAdapter(Config $config): AttributeAdapter
+    {
+        return new AttributeAdapter(
+            $config,
+            $this->getContainer()->get('fin_search.dynamic_product_group'),
+            $this->getContainer()->get(Translator::class),
+            $this->getContainer()->get('fin_search.sales_channel_context'),
+            $this->getContainer()->get(UrlBuilderService::class),
+            $this->getContainer()->get('fin_search.export_context')
+        );
+    }
+
+    /**
+     * @param Attribute[] $attributes
+     * @param array<string, string> $customFields
```

Comment:
This type does not fit for multidimensional custom fields
```suggestion
     * @param array<string, string|array> $customFields
```